api/profile/profile.go

```diff
@@ -359,10 +359,15 @@ func profileFromFile(filePath string) (*Profile, error) {
 	if err != nil {
 		return nil, trace.ConvertSystemError(err)
 	}
-	var p *Profile
+	var p Profile
 	if err := yaml.Unmarshal(bytes, &p); err != nil {
 		return nil, trace.Wrap(err)
 	}
+
+	if p.Name() == "" {
+		return nil, trace.NotFound("invalid or empty profile at %v", filePath)
```

Comment:
Perhaps? It should always be a fairly reasonable path tho.

```suggestion
		return nil, trace.NotFound("invalid or empty profile at %q", filePath)
```